modules/virt-understanding-node-labeling-obsolete-cpu-models.adoc

```diff
@@ -0,0 +1,136 @@
+// Module included in the following assemblies:
+// * virt/node_maintenance/virt-managing-node-labeling-obsolete-cpu-models.adoc
+
+[id="virt-understanding-node-labeling-obsolete-cpu-models_{context}"]
+= Understanding node labeling for obsolete CPU models
+
+To ensure that a node supports only valid CPU models for scheduled VMs, create a ConfigMap with a list of obsolete CPU models. When the `node-labeller` obtains the list of obsolete CPU models, it eliminates those CPU models, and creates labels for valid CPU models.
+
+[NOTE]
+====
+If you do not configure a ConfigMap with a list of obsolete CPU models, all CPU models are evaluated for labels, including obsolete CPU models that are not present in your environment.
+====
+
+The `node-labeller` performs an iterative process to evaluate CPU features in a CPU model specified as the minimum CPU model with other CPU models and their supported CPU features.
+
+Through the process of iteration, the list of base CPU features in the minimum CPU model are eliminated from the list of labels generated for the node. An environment might have two supported CPU models: `Penryn` and `Haswell`.
+
+If `Penryn` is specified as the CPU model for `minCPU`, the `node-labeller` evaluates each base CPU feature for `Penryn` and compares it with each CPU feature supported by `Haswell`. If the CPU feature (for example, `apic`) is supported by both `Penryn` and `Haswell`, the `node-labeller` eliminates `apic` from the list of CPU features for creating labels. If a CPU feature (for example, `aes`) is supported only by `Haswell` and not by `Penryn`, that CPU feature is included in the list of generated labels. The `node-labeller` follows this iterative process to eliminate base CPU features (which are present in the minimum CPU model) and create labels.
+
+The following example shows the complete list of CPU features for `Penryn` which is specified as the CPU model for `minCPU`:
+
+.Example output
+[source,xml]
+----
+apic
+clflush
+cmov
+cx16
+cx8
+de
+fpu
+fxsr
+lahf_lm
+lm
+mca
+mce
+mmx
+msr
+mtrr
+nx
+pae
+pat
+pge
+pni
+pse
+pse36
+sep
+sse
+sse2
+sse4.1
+ssse3
+syscall
+tsc
+----
+
+The following example shows the complete list of CPU features for `Haswell`:
+
+.Example output
+[source,xml]
+----
+aes
+apic
+avx
+avx2
+bmi1
+bmi2
+clflush
+cmov
+cx16
+cx8
+de
+erms
+fma
+fpu
+fsgsbase
+fxsr
+hle
+invpcid
+lahf_lm
+lm
+mca
+mce
+mmx
+movbe
+msr
+mtrr
+nx
+pae
+pat
+pcid
+pclmuldq
+pge
+pni
+popcnt
+pse
+pse36
+rdtscp
+rtm
+sep
+smep
+sse
+sse2
+sse4.1
+sse4.2
+ssse3
+syscall
+tsc
+tsc-deadline
+x2apic
+xsave
+----
+The following example shows the list of node labels generated by the `node-labeler` after iterating and comparing the CPU features for `Penryn` with the CPU features for `Haswell`:
+
+.Example output
+[source,xml]
+----
+aes
+avx
+avx2
+bmi1
+bmi2
+erms
+fma
+fsgsbase
+hle
+invpcid
+movbe
+pcid
+pclmuldq
+popcnt
+rdtscp
+rtm
+sse4.2
+tsc-deadline
+x2apic
+xsave
```

Comment:
It looks like you're missing the ending `----` for this code block. 

I'm guessing that's what's causing the errors that you're seeing in travis. Things like this can be tough to troubleshoot because the errors given by travis come from things later in the build after this.

A tip I'd give would be to take a look at the warnings nearby (the previous one saying `asciidoctor: WARNING: includes/gathering-data-specific-features.adoc: line 135: unterminated listing block` was a clue for me). And especially it's giving errors on things that you did not touch (like the two xref errors), then it's oftentimes something that was caused by some bad syntax prior to those files.

Reply:
Also, you can try doing a local build. If you look at the preview of the local build, you can see that everything after this code block is messed up.

Reply:
Thanks, @bergerhoffer ! That's great info - when I am seeing warnings from other locations, it's related to syntax issues in my file. Local builds - gotta go back to them as well!